webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,257 @@
+import React, {ReactNode} from 'react';
+import './ticketStyle.scss';
+
+import {Dispatch} from 'redux';
+
+import {Instance} from 'types/model';
+import DefaultAvatar from 'components/default_avatar/default_avatar';
+
+export type Props = {
+    href: string;
+    show: boolean;
+    connected: boolean;
+    ticketDetails?: TicketDetails;
+    connectedInstances: Instance[];
+    fetchIssueByKey: (issueKey: string, instanceID: string) => (dispatch: Dispatch, getState: any) => Promise<{
+        data?: TicketData;
+    }>;
+}
+
+export type State = {
+    isLoaded: boolean;
+    ticketId: string;
+    ticketDetails?: TicketDetails;
+};
+
+const isAssignedLabel = ' is assigned';
+const unAssignedLabel = 'Unassigned';
+const jiraTicketTitleMaxLength = 80;
+
+enum myStatus {
+    INDETERMINATE = 'indeterminate',
+    DONE = 'done',
+}
+
+const myStatusClasses: Record<string, string> = {
+    [myStatus.INDETERMINATE]: 'ticket-status--indeterminate',
+    [myStatus.DONE]: 'ticket-status--done',
+};
+
+export default class TicketPopover extends React.PureComponent<Props, State> {
+    constructor(props: Props) {
+        super(props);
+        const {ticketID} = this.getIssueKey();
+
+        this.state = {
+            isLoaded: false,
+            ticketId: ticketID,
+        };
+    }
+
+    getIssueKey = () => {
+        let ticketID = '';
+        let instanceID = '';
+
+        for (const instance of this.props.connectedInstances) {
+            instanceID = instance.instance_id;
+
+            try {
+                if (this.props.href.includes(instanceID)) {
+                    if (this.props.href.includes('selectedIssue=')) {
+                        ticketID = this.props.href.split('selectedIssue=')[1].split('&')[0];
```

Comment:
I think this might be much easier with a regex, and we can do both searches at the same time, but it's probably a nice to have now. It would more readable and easier to understand where it's failing.

Something like this https://regex101.com/r/OxaPmy/1


<img width="1449" alt="Screenshot 2023-01-12 at 09 53 28" src="https://user-images.githubusercontent.com/488556/212021573-a13c13e9-29d0-4884-8a6c-112a624b55ea.png">